sup3r/preprocessing/data_handling.py

```diff
@@ -1814,6 +1858,112 @@ def raw_time_index(self):
                 self._raw_time_index -= pd.Timedelta(12, 'h')
         return self._raw_time_index
 
+    def run_data_extraction(self):
+        """Run the raw dataset extraction process from disk to raw
+        un-manipulated datasets.
+
+        Includes a special method to extract clearsky_ghi from a exogenous
+        NSRDB source h5 file (required to compute clearsky_ratio).
+        """
+        get_clearsky = False
+        if 'clearsky_ghi' in self.raw_features:
+            get_clearsky = True
+            self._raw_features.remove('clearsky_ghi')
+
+        super().run_data_extraction()
+
+        if get_clearsky:
+            cs_ghi = self.get_clearsky_ghi()
+            for it, tslice in enumerate(self.time_chunks):
+                self._raw_data[it]['clearsky_ghi'] = cs_ghi[..., tslice]
+            self._raw_features.append('clearsky_ghi')
+
+    def get_clearsky_ghi(self):
```

Comment:
Should this be put with the other exogenous data handling stuff?

Reply:
you know, i thought about that, but the weirdness is that we want the data handler features to always map to the same set of training features that the model expects. So I thought this was a cleaner solution so that once the data handler is initialized it does indeed have the training and output features that one would expect.

Reply:
ok that's fair